### PR TITLE
Building web application with Kotlin and Typescript

### LINKED ISSUE #4920: Building web application with Kotlin and Typescript
### Topic Suggestion
Building web application with Kotlin and Typescript

### Writing sample(s): 
Include any links or writing samples - to help our team better gauge your writing quality.

#### Proposed title of article
Building web application with Kotlin and Typescript

#### Proposed article introduction
The integration occurs in the context of а web application (bасkend) that runs on а JVM and а browser application (frоntend) that еmрlоys а TypeScriрt-bаsed frаmewоrk like ngulаr оr Reасt (with the TypеScriрtiоn). On the JVM side, Kоtlin integration is simple, and with the right build tool (i.e., Grаdle and the Kоtlin Multiрlаtfоrm Plugin), Kоtlin code is built into a simple jаr mоdule. This JVM JAR can be used just like any other JVM JAR. The JаvаSсriрt side of things is a little more difficult. The build tool will compile the Kotlin code and generate a jаr containing the UMD JаvаSсriрt mоdule (by default). The interesting part is incorporating this into the node. JS-based integrated browser application programming interface

#### Key takeaways 
- The Build Scripts
- The information module
- The client module
- Unpacking the Kotlin modules
- Using Kotlin generated modules in TypeScript code
- TypeScript types for Kotlin code
- TypeScript declarations for third party libraries

#### Article quality
This article describes the problems encountered whilst integrating Kotlin generated modules with a TypeScript browser application and how to overcome them.

#### References
N/A
### Templates to use as guides
- [How To Guide Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/how-to-guide-template.md)
- [Software Review Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/software-review-template.md)
- [Tutorial Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/tutorial-template.md)

### PR DESCRIPTION
**Pull request by 07BLAIZER**
#### Formatting and structure checks
- [X] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**
- [X] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [X] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [X] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**

- [X] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [X] Is the article you are submitting an in-depth and unique article? Does it go beyond what is in the official docs and what is covered in other blog sites. See these articles as [examples](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md).
- [X] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [X] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**

- [X] Have you included a hero image that is **600x400 pixels**, under **300KB in size**, and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
      
#### Grammar & plagiarism checks
- [X] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) - [Grammarly](https://grammarly.com) 
- [X] Please place your article through a 3rd party plagiarism checker. We suggest using Quetext, [this tool](https://www.quetext.com/) is free and has a daily limit. **We typically accept articles with 10% or less.**
- [X] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [ ] Have you added sources for quotes and images that aren't yours?

#### Technical checks
- [X] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**
- [X] Have you checked your code runs correctly and you've highlighted all necessary dependencies for installation?
- [X] Are the software programs and packages you're highlighting in your article up to date, using current versions and not deprecated?

#### Topic suggestion this closes
This closes #4920

### CODE REVIEW

**Comment by ahmadmardeni1** [comment]
Good afternoon and thank you for submitting your article for publishing.
With the recent increase in article volume out team has been working diligently to review every article. Your article has been entered into our queue and should be reviewed (and published pending any changes) as soon as a peer reviewer is finished with the ones in the queue before it. 🚀
Thank you.

**Comment by 07BLAIZER** [comment]
@ahmadmardeni1  hello 👋
Its been 5 days and no peer reviewer has been assigned on this.
Thanks ☺️

**Comment by paulodhiambo** [comment]
@ahmadmardeni1 @07BLAIZER picking up this.

File: content/articles/building-web-application-with-kotlin-and-typescript/index.md [hunk 1]
```diff
@@ -0,0 +1,191 @@
...
+
+The same JVM reflection approach is used, provided that the third party is a Kotlin multiplatform library that includes a JVM target. JVM divides the Kotlin stdlib into several modules while the javascript module provided by Kotlin contains everything thus it becomes crucial to develop its typescript declaration.  
+
+According to the Gradle, the naming conversion is ``` <group>:<name>:<version>``` that is accepted worldwide. Unfortunately, names for both JVM and JavaSript modules do not always match. Thus, it becomes necessary to have code generators for the Typescript declaration that helps in mapping between the Kotlin module and the JVM module. The generators play an important role of correctly adding the imports file to the Typescript declaration. 
+
+### Conclusion
+
+Integration of Kotlin generated modules with Typescript web browser application has been well illustrated in this article. It requires build script codes which are simplified by the use Gradle plugin to set up the Kotlin multiplartform build. After unpacking the KotlinJs modules , it becomes easier to use the Kotlin generated modules into TypeScrits codes of a web application.
```
Comment (paulodhiambo) [inline content/articles/building-web-application-with-kotlin-and-typescript/index.md:189]: Simplify "It requires build script codes which are simplified by the use Gradle plugin..." to "It requires build script codes simplified by the Gradle plugin…".
It should be "multiplatform" not "multiplartform"

File: content/articles/building-web-application-with-kotlin-and-typescript/index.md [hunk 2]
```diff
@@ -0,0 +1,191 @@
...
+
+Dependencies are added to the required Kotlin modules as per the configuration:
+
+```
+//from file: client/build.gradle.kts
+dependencies {
+  nodeKotlin(project(":information"))
+}
+```
+The nodeKotlin dependencies require to be resolved and artifacts that contain the JavaScript codes should be unpacked to the node_modules directory.
```
Comment (paulodhiambo) [inline content/articles/building-web-application-with-kotlin-and-typescript/index.md:138]: rephrase to "The nodeKotlin dependencies require resolving, and we should unpack artifacts containing the JavaScript codes to the node_modules directory."

File: content/articles/building-web-application-with-kotlin-and-typescript/index.md [hunk 3]
```diff
@@ -0,0 +1,191 @@
...
+
+Following is a way of importing generated Kotlin modules to a TypeScript file:
+
+```
+import * as info_js from 'example.addressbook-information';
+import info = info_js.example.addressbook.information;
+...
+let contact = new info.contact('alias')
+```
+import statement on the second time is not much important it also develops a good alias for the content of the modules which helps to reduce more writing of fully qualified names all the time we use Kotlin generated modules.
```
Comment (paulodhiambo) [inline content/articles/building-web-application-with-kotlin-and-typescript/index.md:164]: refrase to "Import statement on the second time is not much important. It also develops a good alias for the content of the modules, which helps to reduce more writing of fully qualified names all the time we use Kotlin generated modules."

**Review by paulodhiambo — CHANGES_REQUESTED** [review]
@07BLAIZER amazing article, make the suggested changes as I proceed with the tech review.

**Comment by lalith1403** [comment]
The plagiarism percentage is way greater than what we typically accept. I will be closing this PR for now. Feel free to make the changes outside of the queue and submit a new PR. 

[Uploading Building web application with Kotlin and Typescript #5466.pdf…]()

**Pull request closed without merging**